### PR TITLE
filter null

### PR DESCRIPTION
**Pull request by jorgeavilacardenosa**

### CODE REVIEW

File: tests/FilterNullsTest.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    /** @test */
+    public function it_should_filter_a_collection_discriminating_nulls()
+    {
+        $actual = ['one' => 1, 'two' => 2, 'three' => null, 'four' => false];
+
+        $this->assertSame(['one' => 1, 'two' => 2, 'four' => false], filter_nulls($actual));
+    }
+
+    /** @test */
+    public function it_should_filter_a_collection_discriminating_nulls_of_a_traversable()
```
Comment (rgomezcasas) [inline tests/FilterNullsTest.php]: I would use a data provider :P

**Comment by eloipoch** [comment]
I were thinking in something more like `filter(not(null()), $collection)` and `filter(not(falsy()), $collection)`. What do you think about it?

**Comment by rgomezcasas** [comment]
@eloipoch We can go with some php... `filter(not('is_null'))`

**Comment by eloipoch** [comment]
@rgomezcasas It seems great to me 👍

File: tests/FilterNullTest.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    public function collections()
+    {
+        return [
+            'array'           => ['actual' => [1, 2, null], 'expected' => [1, 2]],
+            'array_with_keys' => ['actual'   => ['one' => 1, 'two' => 2, 'three' => null, 'four' => false],
+                                  'expected' => ['one' => 1, 'two' => 2, 'four' => false],
+            ],
+            'iterator'        => [
+                'actual'   => new ArrayIterator(['one' => 1, 'two' => 2, 'three' => null]),
+                'expected' => iterator_to_array(new ArrayIterator(['one' => 1, 'two' => 2])),
```
Comment (rgomezcasas) [inline tests/FilterNullTest.php]: Use array directly

**Comment by rgomezcasas** [comment]
:shipit:

**Pull request merged**